src/executor/evaluate/mod.rs

```diff
@@ -197,6 +197,26 @@ pub async fn evaluate<'a, T: 'static + Debug>(
         Expr::Wildcard | Expr::QualifiedWildcard(_) => {
             Err(EvaluateError::UnreachableWildcardExpr.into())
         }
+        Expr::Case {
+            operand,
+            when_then,
+            else_result,
+        } => {
+            let mut wt = Vec::new();
+            for (when, then) in when_then.iter() {
+                wt.push((eval(when).await?, eval(then).await?))
+            }
+
+            let else_result = match else_result {
+                Some(result) => Some(eval(result).await?),
+                None => None,
+            };
+
+            match operand {
+                Some(o) => expr::simple_case(eval(o).await?, wt, else_result),
+                None => expr::searched_case(wt, else_result),
```

Comment:
I checked that `simple_case` and `searched_case` are only being used in here, it means that you don't need to split logic codes into several different ones.
This can simply make code harder to read.

If the reason is that `Case` expression handler codes are too long, then you can split it by making private function like.. `evaluate_case` (ref. `evaluate_function`)
However, it looks just ok to have all in here.

Reply:
Aha that makes sense... I'd better merge those into one `fn case()`. Thanks for feedback!